### PR TITLE
Fix interpodaffinity issue

### LINKED ISSUE #86938: Generate inter pod affinity error
**What happened**:
https://github.com/kubernetes/kubernetes/blob/6099a739d8cb1762cdadf956fae484833b937c5d/pkg/scheduler/framework/plugins/interpodaffinity/filtering.go#L86
Maybe it should be 
```
affinity.PodAffinity != nil
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by bharaththiruveedula-zz**
Signed-off-by: Bharath Thiruveedula <bharath_ves@hotmail.com>

**What type of PR is this?**
> /kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #86938

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
None

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go [hunk 1]
```diff
@@ -1901,6 +1948,7 @@ func TestPreFilterStateAddRemovePod(t *testing.T) {
 			if !reflect.DeepEqual(originalState, state) {
 				t.Errorf("State is not equal, got: %v, want: %v", state, originalState)
 			}
+
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go]: we usually don't add newline like this.

**Review by Huang-Wei — COMMENTED** [review]
Thanks @bharaththiruveedula . LGTM, just one nit.

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go (hunk 1 above)
Comment (bharaththiruveedula) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go]: Done

**Review by draveness — COMMENTED** [review]
/kind bug
/release-note-none

**Comment by draveness** [comment]
Could you please and fill in the release-note section to pass the CI?

**Comment by draveness** [comment]
> Could you please fill in the release-note section to pass the CI?

Never mind. Thanks

**Pull request merged**